lib/spack/spack/fetch_strategy.py

```diff
@@ -318,6 +318,11 @@ def fetch(self):
         url = None
         errors = []
         for url in self.candidate_urls:
+            if url[0:5] == 'azure':
```

Comment:
If `url` is a string it's probably better to:
```suggestion
            if url.startswith('azure'):
```

Reply:
I changed the syntax.